Automattic/wp-calypso
# Global Styles: Show notice in privacy settings

This PR implements the UI changes necessary for displaying a notice in the privacy settings when our users are using Global Styles and are required to upgrade their plan in order to make the customization changes they've made available publicly.

I've decided not to show the notice when the site is not launched because of two main reasons. This isn't specified in the design and I think this wasn't taken into consideration, the second reason is to reduce friction when launching a site, I experimented a bit with the UI and it becomes cluttered with buttons and we aren't providing a clear path that our users should follow, which could lead to a decrease in the number of sites that get launched. 

Also, when a user launches a free site, he will already get prompted to purchase a domain and a plan.

![Captura de Pantalla 2022-10-19 a las 10 12 03](https://user-images.githubusercontent.com/1989914/196634979-ca945c9e-9d59-4de5-a514-affa5643ce85.png)

#### Proposed Changes

* We now check if the user has GS in use and if we should show a notice.
* If the user qualifies, we show a notice that includes a button so that the user can upgrade his plan.

#### Testing Instructions

- [MAKE SURE TO INSTALL THE ETK OF THIS BRANCH](https://github.com/Automattic/wp-calypso/pull/69242#issuecomment-1285541319).
- MAKE SURE YOU ADD THE STICKER BLOG TO THE SITES YOU'LL WORK WITH.

#### Happy Case
* Go to the general settings screen, privacy section. http://calypso.localhost:3000/settings/general/{siteSlug}
* Make sure your site is `FREE` and that you have some GS customizations applied.
   * When the site is not launched you should not see a notice
   * When the site is launched you should see a notice.
      * Make sure the buttons on the notice work as expected.
* Remove or reset your customizations.
* You should not see the notice.

#### Mobile devices
* Repeat the happy case using a mobile viewport (just the part where you interact with the notice).

#### Test un-launched site
* Create a site and make sure to not launch it.
* Make sure it has the corresponding blog sticker.
* Go to the general settings screen, privacy section.
* You should not see the notice.

#### Test tracking event when clicking on the upgrade button.
* Go through the Happy Case
* Open tracks and search for `calypso_global_styles_gating_%`
* You should see some events named `calypso_global_styles_gating_settings_notice_upgrade_click`

#### Test network error when fetching global styles info
* Block the global styles info request in Chrome's inspector and reload.
* You should see no console errors.
* You should see no errors on the screen.
* You should see the last value cached by `useQuery`.

#### Test for regressions
* Accent color should continue to work as before. See this [PR](https://github.com/Automattic/wp-calypso/pull/68841) for testing instructions

### Test evidence
![Captura de Pantalla 2022-10-20 a las 15 27 51](https://user-images.githubusercontent.com/1989914/196961705-6aaa930a-05bf-4f35-be80-14cd4c2d69c1.png)

#### Pre-merge Checklist








Related to #
